phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java

```diff
@@ -228,7 +235,9 @@ public int run(String[] args) throws Exception {
         builder.withSpnego(ugi.getUserName(), additionalAllowedRealms)
             .withAutomaticLogin(keytab)
             .withImpersonation(new PhoenixDoAsCallback(ugi, getConf()));
+
       }
+      setRemoteUserExtractorIfNecessary(builder, getConf());
```

Comment:
With respect to my long-winded comment below, if you're only looking to support Kerberos, we want to move this line into the above if-block.